packages/iris-grid/src/IrisGridMetricCalculator.js

```diff
@@ -8,6 +8,28 @@ import TableUtils from './TableUtils';
  * Call getMetrics() with the state to get metrics
  */
 class IrisGridMetricCalculator extends GridMetricCalculator {
+  getVisibleColumnWidth(
+    column,
+    state,
+    firstColumn = this.getFirstColumn(state),
+    treePaddingX = this.calculateTreePaddingX(state)
+  ) {
+    const { model } = state;
+    const hiddenColumns = model.layoutHints?.hiddenColumns ?? [];
+
+    if (this.userColumnWidths.has(column)) {
```

Comment:
A confusing thing here - this method takes the _visible_ column index as the input, but if you've moved columns around, that will no longer match the _model_ column index.

You need to translate from the visible column index to the model column index. You can do that by calling `const modelColumn = this.getModelColumn(column, state);`. getModelColumn applies the moved column transforms to get the model index.

See `GridMetricCalculator.getVisibleColumnWidth` for an example.